content/docs/buildpack-author-guide/create-buildpack/build-app.md

```diff
@@ -5,9 +5,76 @@ weight=404
 
 <!-- test:suite=create-buildpack;weight=4 -->
 
-Next, we'll make the build step install dependencies. This will require a few updates to the `build` script.
+Now we'll change the build step you created to install application dependencies. This will require updates to the `build` script such that it performs the following steps:
+
+1. Creates a layer for the Ruby runtime
+1. Downloads the Ruby runtime and installs it to the layer
+1. Installs Bundler (the Ruby dependency manager)
+1. Uses Bundler to install dependencies
+
+By doing this, you'll learn how to create arbitrary layers with your buildpack, and how to read the contents of the app in order to perform actions like downloading dependencies.
+
+Let's begin by changing the `ruby-buildpack/bin/build` so that it creates a layer for Ruby.
+
+### Creating a Layer
+
+A Buildpack layer is represented by a directory inside the [layers directory][layers-dir] provided to our buildpack by the Buildpack execution environment. To create a new layer directory representing the Ruby runtime, change the `build` script to look like this:
+
+```bash
+#!/usr/bin/env bash
+set -eo pipefail
+
+echo "---> Ruby Buildpack"
+
+layersdir=$1
+
+rubylayer="$layersdir"/ruby
+mkdir -p "$rubylayer"
+```
+
+The `rubylayer` directory is a sub-directory of the directory provided as the first positional argument to the build script (the [layers directory][layers-dir]), and this is where we'll store the Ruby runtime.
+
+Next, we'll download the Ruby runtime and install it into the layer directory. Add the following code to the end of the `build` script:
+
+```bash
+echo "---> Downloading and extracting Ruby"
+ruby_url=https://s3-external-1.amazonaws.com/heroku-buildpack-ruby/heroku-18/ruby-2.5.1.tgz
+wget -q -O - "$ruby_url" | tar -xzf - -C "$rubylayer"
+```
+
+This code uses the `wget` tool to download the Ruby binaries from the given URL, and extracts it to the `rubylayer` directory.
+
+The last step in creating a layer is writing a TOML file that contains metadata about the layer. Without this file, the Buildpack lifecycle will ignore the layer directory. For the Ruby layer, we need to ensure it is available in the launch image by setting the `launch` key to `true`. Add the following code to the build script:
```

Comment:
Is it worth adding a note that the toml name must match the layer name it is associated with?

Reply:
Fixed in 6da86e3